asterisk_click2dial/models/asterisk_server.py

```diff
@@ -0,0 +1,283 @@
+        help="Company who uses the Asterisk server.",
+    )
+
+    @api.constrains(
+        "out_prefix",
+        "wait_time",
+        "extension_priority",
+        "port",
+        "context",
+        "alert_info",
+        "login",
+        "password",
+    )
+    def _check_validity(self):
+        for server in self:
+            out_prefix = ("Out prefix", server.out_prefix)
+            dialplan_context = ("Dialplan context", server.context)
+            alert_info = ("Alert-Info SIP header", server.alert_info)
+            login = ("AMI login", server.login)
+            password = ("AMI password", server.password)
+
+            if out_prefix[1] and not out_prefix[1].isdigit():
+                raise ValidationError(
+                    _(
+                        "Only use digits for the '%s' on the Asterisk server "
+                        "'%s'" % (out_prefix[0], server.name)
+                    )
+                )
+            if server.wait_time < 1 or server.wait_time > 120:
+                raise ValidationError(
+                    _(
+                        "You should set a 'Wait time' value between 1 and 120 "
+                        "seconds for the Asterisk server '%s'" % server.name
+                    )
+                )
+            if server.extension_priority < 1:
+                raise ValidationError(
+                    _(
+                        "The 'extension priority' must be a positive value for "
+                        "the Asterisk server '%s'" % server.name
+                    )
+                )
+            if server.port > 65535 or server.port < 1:
+                raise ValidationError(
+                    _(
+                        "You should set a TCP port between 1 and 65535 for the "
+                        "Asterisk server '%s'" % server.name
+                    )
+                )
+            for check_str in [dialplan_context, alert_info, login, password]:
+                if check_str[1]:
+                    try:
+                        check_str[1].encode("ascii")
+                    except UnicodeEncodeError:
+                        raise ValidationError(
+                            _(
+                                "The '%s' should only have ASCII caracters for "
+                                "the Asterisk server '%s'" % (check_str[0], server.name)
+                            )
+                        )
+
+    @api.model
+    def _connect_to_asterisk(self):
+        """
+        Open the connection to the Asterisk Manager
+        Returns an instance of the Asterisk Manager
+
+        """
+        user = self.env.user
+        ast_server = user.get_asterisk_server_from_user()
+        # We check if the current user has a chan type
+        if not user.asterisk_chan_type:
+            raise UserError(_("No channel type configured for the current user."))
+
+        # We check if the current user has an internal number
```

Comment:
The comment does not really correspond to the error message, I'd either say `resource` or `internal number`.
Maybe it is clear for pros on Odoo/Asterisk?

Reply:
Agree, I removed the comment, which doesn't add real value.